### PR TITLE
Document new sidecar annotation to exclude interfaces 

### PR DESCRIPTION
**Pull request by obaranov1**
Regarding the added feature istio/istio#34832

### CODE REVIEW

**Comment by ramaraochavali** [comment]
incorrect PR link in the description

**Review by howardjohn — CHANGES_REQUESTED** [review]
I don't see such an annotation anywhere in the codebase?

**Comment by obaranov1** [comment]
the link has been updated. the codebase is there

**Comment by howardjohn** [comment]
But no annotation was ever defined? If you grep the code there is no mention

**Comment by rveerama1** [comment]
We have a use case around this feature. When we were exploring about this feature in the documentation https://istio.io/latest/docs/reference/config/annotations/ we didn't find any thing. Later we noticed about this https://github.com/istio/istio/pull/34832. We haven't tried it on any environment yet. Thought only documentation might be missing. If there are no objections can we proceed with adding pod annotation in the code base?

**Comment by linsun** [comment]
Is annotation better than label here? We are trying to shift away from annotation for sidecar injection.

Can you please document your scenario in a doc as well?

**Comment by howardjohn** [comment]
Annotation is fine here, label is only needed for the webhook stuff since
its used in label selectors

**Comment by rveerama1** [comment]
@howardjohn PTAL about documentation.

**Review by howardjohn — APPROVED** [review]

File: releasenotes/notes/2374.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: documentation
+issue:
+  - https://github.com/istio/api/pull/2374
```
Comment (ericvn) [inline releasenotes/notes/2374.yaml:5]: Nit: Maybe point to feature: https://github.com/istio/istio/pull/34832

**Review by ericvn — APPROVED** [review]
Approved. Marked DNM for the nit. Feel free to fix or not. Then remove the DNM to let this in. Thanks.

**Pull request merged**